clients/slo/slo.go

```diff
@@ -0,0 +1,212 @@
+// @license
+// Copyright 2025 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slo
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+)
+
+const endpointPath = "platform/slo/v1/slos"
+
+func NewClient(client *rest.Client) *Client {
+	c := &Client{
+		restClient: client,
+	}
+	return c
+}
+
+type Client struct {
+	restClient *rest.Client
+}
+
+func (c *Client) List(ctx context.Context) (api.PagedListResponse, error) {
+	var retVal api.PagedListResponse
+
+	listPage := func(ctx context.Context, c *Client, pageKey string) (string, api.ListResponse, error) {
```

Comment:
Why not move this into its own function?

Reply:
It's about style.
The whole function fits to a screen (less than 40 lines of code). The `listPage` is only used inside the `List` method, and they're tightly coupled.

Reply:
I would agree with david it may be that the lines of code are not to many but, do i need to know the details here? Also another point is it provides a better naming possibility.

Reply:
moved